crates/trie/sparse-parallel/src/trie.rs

```diff
@@ -175,23 +193,426 @@ impl ParallelSparseTrie {
         todo!()
     }
 
+    /// Returns the next node in the traversal path from the given path towards the leaf for the
+    /// given full key, or an error if any node along the traversal path is not revealed.
+    ///
+    /// ## Invariants
+    ///
+    /// - `from_path` must be a prefix of `key`.
+    fn find_next_to_leaf(
+        from_path: &Nibbles,
+        from_node: &SparseNode,
+        key: &Nibbles,
```

Comment:
let's use `path` when talking about paths to nodes, and `key` when talking about leaf/extension node key that extends the path to the node to get the full path